commonmark-ext-heading-anchor/src/main/java/ext/heading/anchor/internal/HeaderIdAttributeProvider.java

```diff
@@ -0,0 +1,51 @@
+package ext.heading.anchor.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.commonmark.html.AttributeProvider;
+import org.commonmark.node.AbstractVisitor;
+import org.commonmark.node.Code;
+import org.commonmark.node.Heading;
+import org.commonmark.node.Node;
+import org.commonmark.node.Text;
+
+import ext.heading.anchor.UniqueIdentifierProvider;
+
+public class HeaderIdAttributeProvider implements AttributeProvider {
```

Comment:
This one should be named `HeadingIdAttributeProvider` (the nodes are called `Heading` after the naming was changed in the spec).